tests/plugins/test_push_floating_tags.py

```diff
@@ -0,0 +1,464 @@
+         },
+         'x86_64': {
+             REGISTRY_V2: ['namespace/httpd:worker-build-x86_64-latest'],
+             OTHER_V2: ['namespace/httpd:worker-build-x86_64-latest'],
+         }
+     },
+     None),
+    ("simple_oci",
+     [REGISTRY_V2, OTHER_V2], OCI_RESULTS, 'oci',
+     ['namespace/httpd:release_tag'],
+     {
+         'ppc64le': {
+             REGISTRY_V2: ['namespace/httpd:worker-build-ppc64le-latest'],
+             OTHER_V2: ['namespace/httpd:worker-build-ppc64le-latest'],
+         },
+         'x86_64': {
+             REGISTRY_V2: ['namespace/httpd:worker-build-x86_64-latest'],
+             OTHER_V2: ['namespace/httpd:worker-build-x86_64-latest'],
+         }
+     },
+     None),
+    ("multi_v2",
+     [REGISTRY_V2, OTHER_V2], V2_RESULTS, 'v2',
+     ['namespace/httpd:release_tag', 'namespace/httpd:archive_tag'],
+     {
+         'ppc64le': {
+             REGISTRY_V2: ['namespace/httpd:worker-build-ppc64le-latest'],
+             OTHER_V2: ['namespace/httpd:worker-build-ppc64le-latest'],
+         },
+         'x86_64': {
+             REGISTRY_V2: ['namespace/httpd:worker-build-x86_64-latest'],
+             OTHER_V2: ['namespace/httpd:worker-build-x86_64-latest'],
+         }
+     },
+     None),
+    ("No tags",
+     [REGISTRY_V2, OTHER_V2], V2_RESULTS, 'v2',
+     None,
+     {
+         'ppc64le': {
+             REGISTRY_V2: ['namespace/httpd:worker-build-ppc64le-latest'],
+             OTHER_V2: ['namespace/httpd:worker-build-ppc64le-latest'],
+         },
+         'x86_64': {
+             REGISTRY_V2: ['namespace/httpd:worker-build-x86_64-latest'],
+             OTHER_V2: ['namespace/httpd:worker-build-x86_64-latest'],
+         }
+     },
+     'No floating images to tag, skippping push_floating_tags'),
+    ("called_from_worker",
+     [REGISTRY_V2, OTHER_V2], V2_RESULTS, 'v2',
+     ['namespace/httpd:release_tag', 'namespace/httpd:archive_tag'],
+     {},
+     'push_floating_tags cannot be used by a worker builder'),
+    ("No_results",
+     [REGISTRY_V2, OTHER_V2], None, 'oci',
+     ['namespace/httpd:release_tag', 'namespace/httpd:archive_tag'],
+     {
+         'ppc64le': {
+             REGISTRY_V2: ['namespace/httpd:worker-build-ppc64le-latest'],
+             OTHER_V2: ['namespace/httpd:worker-build-ppc64le-latest'],
+         },
+         'x86_64': {
+             REGISTRY_V2: ['namespace/httpd:worker-build-x86_64-latest'],
+             OTHER_V2: ['namespace/httpd:worker-build-x86_64-latest'],
+         }
+     },
+     'No manifest list digest available, skipping push_floating_tags'),
+])
+@responses.activate  # noqa
+def test_floating_tags_push(tmpdir, test_name, registries, manifest_results, schema_version,
+                            floating_tags, workers, expected_exception, reactor_config_map,
+                            caplog):
+    if MOCK:
+        mock_docker()
+
+    test_images = ['namespace/httpd:2.4', 'namespace/httpd:latest']
```

Comment:
These are intended to be images with 'primary' tags. To make the test less confusing to read can we use something like 2.4-1 instead of 2.4 and latest (which are both floating tags)?
```suggestion
    test_images = ['namespace/httpd:2.4-1']
```

Reply:
I'm renaming it to `primary_tags` ; the actual test tags are passed as arguments to the test function.

Reply:
Thanks!